src/openrct2/actions/LargeSceneryRemoveAction.hpp

```diff
@@ -0,0 +1,251 @@
+
+public:
+    LargeSceneryRemoveAction() = default;
+
+    LargeSceneryRemoveAction(int16_t x, int16_t y, uint8_t baseHeight, uint8_t direction, uint16_t tileIndex)
+        : _x(x)
+        , _y(y)
+        , _baseHeight(baseHeight)
+        , _direction(direction)
+        , _tileIndex(tileIndex)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
+    }
+
+    void Serialise(DataSerialiser & stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << DS_TAG(_x) << DS_TAG(_y) << DS_TAG(_baseHeight) << DS_TAG(_direction) << DS_TAG(_tileIndex);
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        GameActionResult::Ptr res = std::make_unique<GameActionResult>();
+
+        const uint32_t flags = GetFlags();
+
+        int32_t z = tile_element_height(_x, _y);
+        res->Position.x = _x + 16;
+        res->Position.y = _y + 16;
+        res->Position.z = z;
+        res->ExpenditureType = RCT_EXPENDITURE_TYPE_LANDSCAPING;
+        res->Cost = 0;
+
+        TileElement* tileElement = FindLargeSceneryElement();
+        if (tileElement == nullptr)
+        {
+            log_warning("Invalid game command for scenery removal, x = %d, y = %d", _x, _y);
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_SELECTION_OF_OBJECTS);
+        }
+
+        rct_scenery_entry* scenery_entry = tileElement->AsLargeScenery()->GetEntry();
+
+        LocationXYZ16 firstTile = {
+            scenery_entry->large_scenery.tiles[_tileIndex].x_offset, scenery_entry->large_scenery.tiles[_tileIndex].y_offset,
+            static_cast<int16_t>((_baseHeight * 8) - scenery_entry->large_scenery.tiles[_tileIndex].z_offset)
+        };
+
+        rotate_map_coordinates(&firstTile.x, &firstTile.y, _direction);
+
+        firstTile.x = _x - firstTile.x;
+        firstTile.y = _y - firstTile.y;
+
+        bool calculate_cost = true;
+        for (int32_t i = 0; scenery_entry->large_scenery.tiles[i].x_offset != -1; i++)
+        {
+            LocationXYZ16 currentTile = { scenery_entry->large_scenery.tiles[i].x_offset,
+                                          scenery_entry->large_scenery.tiles[i].y_offset,
+                                          scenery_entry->large_scenery.tiles[i].z_offset };
+
+            rotate_map_coordinates(&currentTile.x, &currentTile.y, _direction);
+
+            currentTile.x += firstTile.x;
+            currentTile.y += firstTile.y;
+            currentTile.z += firstTile.z;
+
+            if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) && !gCheatsSandboxMode)
+            {
+                if (!map_is_location_owned(currentTile.x, currentTile.y, currentTile.z))
+                {
+                    return MakeResult(GA_ERROR::NO_CLEARANCE, STR_CANT_REMOVE_THIS, STR_LAND_NOT_OWNED_BY_PARK);
+                }
+            }
+
+            // NOTE: Not sure what to make of this, is this actually correct?
```

Comment:
This is tied to `ResetClearLargeSceneryFlag`. It prevents a double+ cost when the clear scenery tool overlaps multiple tiles of a large scenery element. I've changed the comment in my quarter tile refactor. #8722